test/functional/fixtures/references/references_within_references/input.json

```diff
@@ -0,0 +1,47 @@
+{
+	"name": "ARGO Dictionary",
+	"version": "1.2",
+	"schemas": [
+		{
+			"name": "donor",
+			"description": "Donor Entity",
+			"fields": [
+				{
+					"name": "donor_submitter_id",
+					"displayName": "Submitter Donor ID",
+					"valueType": "string",
+					"description": "Unique identifier for donor; assigned by data provider",
+					"meta": {
+						"key": true
+					},
+					"restrictions": {
+						"regexp": "#/ID_REG_EXP"
+					}
+				},
+				{
+					"name": "gender",
+					"valueType": "string",
+					"description": "Donor Biological Sex",
+					"restrictions": {
+						"codeList": ["#/SEX", "Other"]
+					}
+				},
+				{
+					"name": "ethnicity",
+					"valueType": "string",
+					"description": "Self described",
+					"meta": {
+						"default": "Unknown"
+					},
+					"restrictions": {}
+				}
+			]
+		}
+	],
+	"references": {
+		"ID_REG_EXP": "^[\\w\\s\\W]{5,}$",
+		"MALE": "Male",
+		"FEMALE": "Female",
+		"SEX": ["#/MALE", "#/FEMALE"]
```

Comment:
Did we cover the case where the nested reference is also an array?

```json
"LETTERS": ["A","B"],
"NESTED_REFERENCE": ["#/LETTERS", "C"]
```

Reply:
Changed input.json to reflect this scenario.
```
"UNKNOWN": ["Unknown", "No answer"],
"SEX": ["#/MALE", "#/FEMALE", "#/UNKNOWN"]
```